src/lib/storage/table.hpp

```diff
@@ -87,5 +88,8 @@ class Table : private Noncopyable {
   uint32_t _max_chunk_size;
 
   void _append_new_chunk();
+
+  static void _compress_segment(std::promise<std::shared_ptr<BaseSegment>> promise, const std::string type,
```

Comment:
Should uncompressed_segment be a reference?

Reply:
we're already passing by pointer, so I think this is fine. It's a completely different execution context, so I think it's worth the ref-counting overhead.